airbyte-integrations/connectors/source-mixpanel/source_mixpanel/source.py

```diff
@@ -812,9 +812,7 @@ def streams(self, config: Mapping[str, Any]) -> List[Stream]:
         start_date = config.get("start_date")
         if start_date and isinstance(start_date, str):
             start_date = pendulum.parse(config["start_date"]).date()
-        year_ago = now - timedelta(days=365)
-        # start_date can't be older than 1 year ago
-        config["start_date"] = start_date if start_date and start_date >= year_ago else year_ago  # set to 1 year ago by default
+        config["start_date"] = start_date or now
```

Comment:
why is this set to now? Shouldn't it be a time in the past?

Reply:
`start_date` is not required in spec file. So if `start_date` is not specified by the user we will use default value which is `now` (same for `end_date`).

Reply:
wouldn't `now` sync no records? The linked issue is saying that if `start_date` is not specified, then all data should be synced from the mixapenl account

Reply:
> wouldn't `now` sync no records?

Nope, it will sync records for `now`.
<img width="741" alt="image" src="https://user-images.githubusercontent.com/19872253/141023768-d7d9a800-259a-45b6-b202-4f611926447e.png">


> The linked issue is saying that if start_date is not specified, then all data should be synced from the mixapenl account

Sorry, I really don't see where in #7451 this is written 😢

Reply:
the current logic `config["start_date"] = start_date or now` means that if the user did not provide an input date, we start syncing their data from today. Meaning any data that was generated 6 months ago wouldn't get synced. I think this is not the desirable behavior. 

I think the right behavior here should be: if user specifies a start_date, use that. Otherwise sync from 1 year ago i.e: `config["start_date"] = start_date or _one_year_ago`

Reply:
> I think the right behavior here should be: if user specifies a start_date, use that. Otherwise sync from 1 year ago i.e: `config["start_date"] = start_date or _one_year_ago`

Okay, but why not 2 year ago, for example, or three? I mean why exactly 1 year ago?

Reply:
Fair question. I think it's less important what the default is exactly, but I suspect for many users, analyzing data over a year old isn't going to be very common. So honestly, the 1 year is just a hunch. maybe the right number is 6 months or 2 years, I'm not sure. I think the important part is a user can set the start_date if they like to something older than 1 year

Reply:
Ok, got it :)